### PR TITLE
Change ABI compatibility report from HTML to text

### PR DESCRIPTION
**Pull request by jmcatamney**

### CODE REVIEW

**Comment by Eulerizeit** [comment]
Is there an example we can look at?

**Comment by jchampio** [comment]
I think the original intent was to have this text dumped into the output of the Concourse job itself. @skahler-pivotal ?

**Comment by jmcatamney** [comment]
@Eulerizeit: There's one at s3://gpdb-public-web-dev/compat_reports/postgres/5.7.0_to_HEAD/compat_report.txt (can't link it inline, sorry) produced by the story pipeline.

@jchampio: Ah, if by "output" the story meant messages and not job outputs, I can modify it to do that too.

**Comment by jmcatamney** [comment]
@Eulerizeit @jchampio Take a look at the `compare_abi` task at https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-abi/jobs/compile_gpdb_ubuntu16_oss_abi/builds/6 and see if that looks good to you.

**Comment by jchampio** [comment]
@jmcatamney To me it's a little weird that the script now dumps all the reports during the "these pieces failed" output, instead of only dumping the pieces that failed. But I am not the intended audience, and if the PMs are happy then I am happy.

**Comment by jmcatamney** [comment]
@jchampio Whoops, for some reason I was thinking it did only generate reports for the failed binaries.  You're right, as-is they don't make sense there.

@Eulerizeit @skahler-pivotal @bharath1028:
Would it be more useful from a PM standpoint to dump every report every run in their own section, or to only dump runs that fail on a given run in the "these binaries failed" section?

**Comment by Eulerizeit** [comment]
On Wed, Apr 24, 2019 at 1:36 PM Jamie McAtamney <notifications@github.com>
wrote:

> @jchampio <https://github.com/jchampio> Whoops, for some reason I was
> thinking it did only generate reports for the failed binaries. You're
> right, as-is they don't make sense there.
>
> @Eulerizeit <https://github.com/Eulerizeit> @skahler-pivotal
> <https://github.com/skahler-pivotal> @bharath1028
> <https://github.com/bharath1028>:
> Would it be more useful from a PM standpoint to dump every report every
> run in their own section, or to only dump runs that fail on a given run in
> the "these binaries failed" section?
>

WRT what I would like to see. What you have is fine, however, the report
should be from the last tagged version to HEAD on the branch.

I think the end goal, and certainly beyond the scope of this story, is to
decide what ABI changes we can except (non-shipping components such as
gp_fault_injector) and then ignore those and for any ABI we do care about,
explicitly have a red test for those so we can make sure that the change is
necessary/unavoidable.

-- Rob

**Comment by jmcatamney** [comment]
@bharath1028 @Eulerizeit @skahler-pivotal @jchampio

The task now dumps only reports for the failed binaries; see the `compare_abi` output [here](https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-abi/jobs/compile_gpdb_ubuntu16_oss_abi/builds/16).  If that looks good to all y'all, we'll merge this later today.

@Eulerizeit 
> What you have is fine, however, the report should be from the last tagged version to HEAD on the branch.

This branch just had stale tags when we branched and Concourse didn't pick it up when we pushed tags.  5X_STABLE is using the latest version in its ABI job, so this should use the correct tags when it's merged.

File: concourse/scripts/compare_abi_ubuntu.bash [hunk 1]
```diff
@@ -38,25 +36,32 @@ make -C abi-compliance-checker install prefix=/usr/local
...
     echo "The following binaries have ABI differences from $old:"
     echo
-    echo "    $failed"
+    echo "    ${failed_binaries[*]}"
     echo
-    echo "A report has been generated."
+    echo "A report has been generated, and is displayed below:"
+    for failed_binary in ${failed_binaries[@]}; do
+        # directory structure is /path/[binary name]/X.Y.Z_to_HEAD/compat_report.txt
+        report_path="./abi/compat_reports/$(basename $failed_binary)/${old}_to_HEAD/compat_report.txt"
```
Comment (jchampio) [inline concourse/scripts/compare_abi_ubuntu.bash]: `HEAD` should be `${new}` here to help future maintainers with the commit-to-commit reporting that has been asked for but not yet implemented.

Comment (jmcatamney) [reply concourse/scripts/compare_abi_ubuntu.bash]: 👍

**Pull request merged**